Editor/Scripts/CodeGenerators/GenerateRGStateClasses.cs

```diff
@@ -0,0 +1,276 @@
+using Newtonsoft.Json;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using System;
+using UnityEditor;
+using UnityEngine;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+namespace RegressionGames
+{
+    public static class GenerateRGStateClasses
+    {
+        public static void Generate(string jsonInput)
```

Comment:
Is there an example in some repo of the generated file?

Reply:
PR'd in Unity Sample with generated state file: https://github.com/Regression-Games/RGUnitySample/pull/8